### ISSUE TITLE
foobar_2-2_power_hungry.py - wrong logic

### ISSUE THREAD

**Issue by DmitryZayats** [FoxHub/Google-FooBar#2]
This code

 ` if (npos == 0 and nneg == 1) or (npos == 0 and nneg == 0):
        return xs[0]`

behaves differently based on exact sequence of numbers in the list.

If you add to test cases these lines

    case6 = [0,-2]
    print "Case 6 : ", answer(case6)
    case7 = [-2,0]
    print "Case 7 : ", answer(case7)

Output will be different.

    Case 6 :  0
    Case 7 :  -2

Logically from changing sequence of numbers in input array - answer should not change. 
Logically speaking if we have array where some panels have 0 output and one has drain (negative) - the max power is 0, not negative.
But your code is the only way to pass test cases 3 and 4, since one of them expects 0 and the other one expects negative (depending on sequence of this negative in the list, which doesn't make any sense).

**Comment by FoxHub** [comment]
Yep, you're correct. This logic is bad, but it's required to pass the test, because the challenge itself expects that the following are true:

> case8 = [-16]
> print "Case 8 : " answer(case8)
> case9 = [-16, 0, 0, 0, 0]
> print "Cast 9 : " answer(case9)

Would output:

> Case 8 : -16
> Case 9 : -16

There's some logic on the validator's end that seems to believe the maximum product of a sequence beginning with a negative number and then followed by zeroes is going to be the negative number, rather than zero (so in our imaginary case 9, `16`.)

What this could should do, if it were solving a real problem, would be to have a sub-case that returns zero like the following:

`if (nneg == 1 and npos == 0 and nzero > 0): return 0`

... or something similar. You get the point.
